src/components/ResourceProvider.tsx

```diff
@@ -59,8 +59,8 @@ class ResourceProviderBase<T> extends React.Component<ResourceProviderProps<T>,
   }
 
   private dispatchRequest(props: ResourceProviderProps<T>) {
-    if (!props.request || !props.request.status || this.props.resourceLocator.url !== props.resourceLocator.url)
-      props.dispatchRequest(props.resourceLocator)
+    const { request } = props
```

Comment:
Why is this change needed / what is it doing?

Reply:
because the request is `undefined` sometimes, this code prevents the crash in the `console`

Reply:
And why is the `this.props.resourceLocator.url !== props.resourceLocator.url` removed? I'm not sure why it was there to being with, but I suspect it's preventing the trigger of new download requests for re-renders of the component when some unrelated prop changed

Reply:
for the moment with the new condition, all scenarios are controlled. sincerely I don't know why we should keep the before the condition